pandas/core/strings.py

```diff
@@ -486,24 +489,103 @@ def f(x):
         else:
             return empty_row
 
-    if regex.groups == 1:
-        result = np.array([f(val)[0] for val in arr], dtype=object)
-        name = _get_single_group_name(regex)
+    names = dict(zip(regex.groupindex.values(), regex.groupindex.keys()))
+    columns = [names.get(1 + i, i) for i in range(regex.groups)]
```

Comment:
this is annoying, but prob should have a test that verifies when you have *some* named groups and others that are not named, you would get columns like ``Index(['foo',1,'bar',3])``.